src/main/resources/avro/variants.avdl

```diff
@@ -178,42 +232,181 @@ record Variant {
-  This corresponds to the first base of the string of reference bases.
-  Genomic positions are non-negative integers less than reference length.
-  Variants spanning the join of circular genomes are represented as
-  two variants one on each side of the join (position 0).
+  The name of the call set this `Call` belongs to.
   */
-  long start;
+  union { null, string } callSetName = null;
 
   /**
-  The end position (exclusive), resulting in [start, end) closed-open interval.
-  This is typically calculated by `start + referenceBases.length`.
+  The ID of the `Allele` that this `Call` reflects the presence of.
   */
-  long end;
+  string alleleId;
 
-  /** The reference bases for this variant. They start at the given position. */
-  string referenceBases;
+  /**
+  The ID of the `Variant` in the context of which this `Call` is calling the
+  copy number on the `Allele`. The `Allele` called by this `Call` must be on the
+  list of `Allele` IDs that are allowable alternatives for the `Variant`. Only
+  one `Call` in a `CallSet` can describe a given `Allele` within a particular
+  `Variant`.
+  */
+  union { null, string } variantId = null;
+
+  // HAPLOTYPE-type CallSets double as phase sets, so we don't need phase sets.
 
-  /** The bases that appear instead of the reference bases. */
-  array<string> alternateBases = [];
+  // We have various ways of specifying what the call actually is.
+  // Not all of the following will be present, but at least one should be
 
   /**
-  A map of additional variant information.
+  0 for not present in callSet, 1 for single copy, 2 for 2 copies etc.
+  */
+  union { null, int } copyNumber = null;
+
+  /**
+  Expected value of copyNumber.
+  */
+  union { null, float } dosage = null;
+
+  /**
+  Probability that copy number is k: 0..n
+  */
+  array<float> probability = [];
+
+  /**
+  Likelihood of data given copy number k.
+  */
+  array<float> likelihood = [];
+
+  /**
+  Number of reads supporting this allele (DP in VCF).
+  */
+  union { null, int } readCount = null;
+
+  /**
+  Potential other evidence for the call.
+  */
+  map<string> evidence = {};
+
+  /**
+  A map of additional allele call information.
   */
   map<array<string>> info = {};
+}
 
+/**
+`Allele`s are the key component of the "allelic" model of variation: they are
+single objects that are present in some copy number.
+
+An `Allele` is a contigous piece of sequence that we will want to say is present
+(in general at some copy number) or absent in a sample. Very often it will just
+be a single segment covering the entirety of a novel `Reference`, but it is also
+common for it to be a `Segment` of a `Reference` not overlapped by another
+`Allele`'s `Segment`s, or in general to be any contiguous path through the
+sequence graph of `Reference`s.
+
+`Allele`s belong to `VariantSet`s.
+
+Remember that `Path`s cannot follow adjacencies not represented in the
+`Reference`s they use. If you want an `Allele` that spans a novel deletion, say,
```

Comment:
Again, I think this is too informal for an API specification. I would say, "Note that `Path`s cannot follow adjacencies not represented in the `Reference`s they use. If  an `Allele` that spans a novel deletion is required, for example, a new `Reference` must be included in the `VariantSet` providing that adjacency, which this `Allele` can then follow."